src/serial.rs

```diff
@@ -186,16 +186,53 @@ unsafe impl<USART> Send for Tx<USART> {}
 
 #[cfg(feature = "device-selected")]
 macro_rules! usart {
-    ($($USART:ident: ($usart:ident, $usartXen:ident, $apbenr:ident),)+) => {
+    ($($USART:ident: ($usart:ident, $usarttx:ident, $usartrx:ident, $usartXen:ident, $apbenr:ident),)+) => {
         $(
             use crate::stm32::$USART;
-            impl<TXPIN, RXPIN> Serial<$USART, TXPIN, RXPIN> {
+            impl<TXPIN, RXPIN> Serial<$USART, TXPIN, RXPIN>
+            where
+                TXPIN: TxPin<$USART>,
+                RXPIN: RxPin<$USART>,
+            {
                 /// Creates a new serial instance
                 pub fn $usart(usart: $USART, pins: (TXPIN, RXPIN), baud_rate: Bps, clocks: Clocks) -> Self
-                where
-                    TXPIN: TxPin<$USART>,
-                    RXPIN: RxPin<$USART>,
                 {
+                    let mut serial = Serial { usart, pins };
+                    serial.enable(baud_rate, clocks);
+                    serial
+                }
+            }
+
+            impl<TXPIN> Serial<$USART, TXPIN, ()>
+            where
+                TXPIN: TxPin<$USART>,
+            {
+                /// Creates a new tx-only serial instance
+                pub fn $usarttx(usart: $USART, txpin: TXPIN, baud_rate: Bps, clocks: Clocks) -> Self
+                {
+                    let rxpin = ();
+                    let mut serial = Serial { usart, pins: (txpin, rxpin) };
+                    serial.enable(baud_rate, clocks);
+                    serial
```

Comment:
Why not have the Tx only constructor just return a `Tx<USART>`?

Reply:
I want to enable `listen` and potential future things for Rx/Tx only usarts. Returning a `Tx` would also make `release` pretty hard to do.

Reply:
I don't see how that fixes split usarts from being unable to release their pins?

Reply:
That was in reference to the second part. The first part is also party addressed by just not needing to split, but to release the pins of the split usarts would be a bit more involved. We'd need to change `Tx` & `Rx` to contain pin references and also add an api to create a pin instance from a function (which is a bit out of scope for this pr).

Reply:
I suppose I tend to see splitting usarts as a useful and important feature as it allows them to be partially shared into interrupt handlers in a sensible way without lock contention.

Reply:
That sounds reasonable to me, also I'm unsure over the use of generating temporary Tx/Rx instances to impl the Read/Write traits, it would seem more appropriate to move the implementations out to free functions that take a *const USART argument.

Reply:
Moved it out in separate functions with RegisterBlock, I could remove the USART parameter from `Tx` & `Rx` type, not sure if I should.

Reply:
I'd leave it there, its potentially relevant to distinguish Tx and Rx objects from different USART types.